common/djangoapps/track/utils.py

```diff
@@ -28,3 +30,38 @@ def default(self, obj):  # pylint: disable=method-hidden
             return obj.isoformat()
 
         return super(DateTimeJSONEncoder, self).default(obj)
+
+
+class LatinJSONEncoder(DateTimeJSONEncoder):
+    """
+    JSON encoder for UnicodeDecodeError with latin1 characters
+    """
+
+    def encode(self, obj):
+        """
+        encode method gets an original object
+        and returns result string. obj argument will be the
+        object that is passed to json.dumps function
+        """
+        obj = self.iterate_dictionary(obj)
+
+        return super(LatinJSONEncoder, self).encode(obj)
+
+    def iterate_dictionary(self, obj):
+        """
+        Iterate over nested dict structure
+        """
+        for key, value in obj.iteritems():
+            if isinstance(value, dict):
+                self.iterate_dictionary(value)
+            elif isinstance(value, str):
```

Comment:
Sure @adampalay, so for more logging instead of only logged effected event data (current) i will log the whole event for better information.

@brianhw should i also logged the traceback of Exception ?